src/tyr/locate_serializer.cc

```diff
@@ -1,11 +1,57 @@
 #include "baldr/json.h"
+#include "baldr/openlr.h"
 #include "tyr/serializers.h"
 #include <cstdint>
 
 using namespace valhalla;
+using namespace valhalla::midgard;
 using namespace valhalla::baldr;
 
 namespace {
+
+OpenLR::LocationReferencePoint::FormOfWay get_fow(const baldr::DirectedEdge* de) {
```

Comment:
Curious what you think about taking a reference instead of a pointer as argument? It doesn't seem like it should be possible to send a nullpointer so why not rule out that possibility entirely?

Reply:
the are essentially the same thing. the only reason i didnt do it is because its not a reference when we get it from graphreader. to turn it into a reference we have to derefrence it (ie an extra step with no benefit). basically pretty much all the other code just uses const pointers for the native data and so this fits the pattern. now that im t hinking of it though, maybe it would be cool to make this a member function of directededge? though that kind of scatters the code around a bit..